### PR TITLE
[docs] Improve instructions for using Charles to debug spaceship

### PR DESCRIPTION
**Pull request by janpio**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
ToolsAndDebugging.md and the spaceship docs had duplicated information and made it harder than necessary for someone to link to one location to instruct people on how to use Charles to create a traffic session.

### Description
Move the technical stuff to spaceship, crosslink back from spaceship to the existing Charles instructions. Fix tiny things along the way.

### CODE REVIEW

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Pull Request is Work in Progress</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: ToolsAndDebugging.md [hunk 1]
```diff
@@ -28,45 +28,11 @@ DEBUG=1 bundle exec rspec
...
-- `https://developer.apple.com`:
-  - List all devices, certificates, apps and app groups
-  - Create new certificates, provisioning profiles and apps
-  - Disable/enable services on apps and assign them to app groups
-  - Delete certificates and apps
-  - Repair provisioning profiles
-  - Download provisioning profiles
-  - Team selection
-- `https://appstoreconnect.apple.com`:
-  - Managing apps
```
Comment (KrauseFx) [inline ToolsAndDebugging.md]: Did you remove those sub-lists on purpose?

**Review by KrauseFx — APPROVED** [review]
Solid 👍

**Review by KrauseFx — APPROVED** [review]

**Comment by janpio** [comment]
🚀

**Pull request merged**